### ISSUE TITLE
Mac issue: vifm unable to fork commands like !mkdir and !touch or even open files with a text editor

### ISSUE THREAD

**Issue by ngriffin7a** [vifm/vifm#884]
When I run vifm from Terminal.app on a Mac running Ventura, I am able to navigate the directory structure, but unable to do anything that would (I assume) fork another command.

For example, if I run `:mkdir foo` it reports:
0 directories created

If I run `!touch foo.txt` it fails silently.

If I try to open a text file, the screen quickly blinks and the text editor doesn't appear.

Now, if I run vifm with sudo it works, but then everything is owned by root, so that doesn't help.

I checked and I am able (with Terminal.app), to execute `mkdir` and `touch` directly from the command line without `sudo`

Finally, I did a `tail -f $HOME/.vifm/log` but it doesn't report any error messages in the log.

I checked over-and-over, and the folder structure has correct filesystem permissions.

In addition, Terminal.app has been granted the "Full Disk Access" security permission.

Anyone have any thoughts as to what might be wrong with my environment? Or how I can get more verbose error messages from vifm?

Thanks in advance.

**Comment by ngriffin7a** [comment]
Update: If I `rm -rf ~/.vifm` then vifm seems to work OK.

If I create an empty ~/.vifm directory, then run vifm, it has the same problematic behavior as in the issue description.

**Comment by xaizek** [comment]
> Update: If I `rm -rf ~/.vifm` then vifm seems to work OK.

Including `:!touch bla`? If not, then it's just the effect of `set syscalls`.

In case that it doesn't work due to some other settings mind that:
 * if you remove `~/.vifm/`, but `~/.config/` exists, Vifm will create/use `~/.config/vifm/` instead
 * creating empty `~/.vifm/` makes Vifm use default values for all options

You can run `:set` without arguments when things work to see options that differ from the defaults.

> Finally, I did a `tail -f $HOME/.vifm/log` but it doesn't report any error messages in the log.

Were you running `vifm --logging`? Logging is off by default.

**Comment by ngriffin7a** [comment]
@xaizek thank you so much for the response! :) 

When I specify `syscalls` the internal `:mkdir` starts to work.

However, calling external commands such as `:!mkdir foo` or `:!touch foo.txt` or even opening a file like foo.pdf by pressing return/enter does not work. The only way I can make external commands work is through `sudo vifm`

When I try `vifm --logging` I don't see any error output when I try to invoke those external commands I just mentioned.

I'd appreciate any further insights you might have. Thanks again.

**Comment by xaizek** [comment]
> The only way I can make external commands work is through `sudo vifm`

Running as root can make Vifm use a different configuration directory. See output of `:echo $VIFM`.

> In addition, Terminal.app has been granted the "Full Disk Access" security permission.

Maybe some security settings that don't apply to root has something to do with this. I wouldn't expect rights of a terminal to affect rights of applications run in the terminal.

> However, calling external commands such as `:!mkdir foo` or `:!touch foo.txt` or even opening a file like foo.pdf by pressing return/enter does not work.

Was this said about `set syscalls` (which has no effect on `:!`) or `rm -rf ~/.vifm`?

**Comment by ngriffin7a** [comment]
When I run `vifm` as the ngriffin user, `:echo $VIFM` outputs `/Users/ngriffin/.vifm`

When I run `sudo vifm`, `:echo $VIFM` outputs the same `/Users/ngriffin/.vifm`

Regarding execution of external commands using `:!`, the only way I can get that to work is with `sudo`. There is no scenario as the ngriffin user that makes it work. The `~/.vifm` directory can be present, or not present, execution of external commands does not work.

Again, thanks so much for your help. Appreciate your time. I have grown very dependent on `vifm` and miss having it! It's definitely an old friend. :-)

**Comment by xaizek** [comment]
Thanks for the clarification. So did it stop working after an upgrade to Ventura? It sounds like it worked for you before or was it on Linux?

**Comment by ngriffin7a** [comment]
No, I actually have another Mac right next to it where vifm works fine, and it's running Ventura too.

They are both Intel MacBooks from the same year with almost identical hardware. The only difference that I am aware of, is that the MacBook where vifm is not working has [FileVault](https://support.apple.com/en-us/HT204837) enabled (encrypted filesystem), but the MacBook where vifm does indeed work does not have FileVault enabled (non-encrypted filesystem).

But I don't think FileVault on/off is related, because vifm doesn't work on a non-encrypted USB drive with an ExFAT filesystem either.

**Comment by ngriffin7a** [comment]
In order to provide a more complete answer to your question, both MacBooks started with a fresh install of Ventura, not upgrades to Ventura.

**Comment by ngriffin7a** [comment]
Update, I temporarily moved my .zshrc and .bashrc files elsewhere, and vifm started working properly! :) I will attempt to figure out what's causing the problem and report back, in case someone else runs into this problem in the future.

**Comment by ngriffin7a** [comment]
Found it! :-)

My $HOME/.zshrc had the following on line#1:

`export SHELL="/usr/local/bin/zsh"`

So I must have done `brew install zsh` on the MacBook where vifm was working.

But on the MacBook where vifm was not working, I didn't have zsh on that path. When I changed it to the following (the default location for zsh that ships with MacOs), it started working:

`export SHELL="/bin/zsh"`

Thanks again for the help and support! :-)
